examples/solvers/nuclear_norm_tomography.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tomography with Nuclear norm regularization .
+
+Solves the optimization problem
+
+    min_{0 <= x_1 <= 1, 0 <= x_2 <= 1}
+        ||Ax_1 - g_1||_2^2 + 0.1 ||Ax_2 - g_2||_2^2 +
+        lam || [grad(x_1), grad(x_2)] ||_*
+
+Where ``A`` is the ray transform,  ``grad`` is the spatial gradient,
+``g_1``, ``g_2`` the given noisy data and ``|| . ||_*`` is the nuclear-norm.
+
+The nuclear norm takes a vectorwise matrix norm, the spectral norm, i.e. the
+p-norm of the singular vectors. It introduces a coupling between the terms
+which allows better reconstructions.
+
+In this case we assume that ``g_2`` is much more noisy than ``g_1``, but we can
+still get an acceptable reconstruction.
+
+The data is assumed to be similar, but not exactly the same, for the channels.
+
+Note that this is an advanced example.
```

Comment:
I don't know... if we had a place to put this advanced stuff, this example would fit. Why not keep it for now, I feel like not all examples have to be dead simple.